lib/matplotlib/axes/_axes.py

```diff
@@ -5761,6 +5762,8 @@ def hist(self, x, bins=10, range=None, normed=False, weights=None,
             for m, c in zip(n, color):
                 if bottom is None:
                     bottom = np.zeros(len(m), np.float)
+                if input_empty:
+                    width = 0
```

Comment:
As per @efiring 's comment: "I think we should follow that logic, which implies that we should be drawing zero-height (or width) patches." 

Without setting width to 0, the patches for bar have size 0.1 and thus an empty graph (as desired) is not drawn.

Before (width 0.1), non empty graph:
![bar_width_nonzero](https://cloud.githubusercontent.com/assets/5559279/6648300/a62a67e6-c9ae-11e4-947d-206f4f2d1b1d.png)

After (width 0, empty graph):
![bar_width_zero](https://cloud.githubusercontent.com/assets/5559279/6648299/a62123de-c9ae-11e4-991f-18075dce8bbd.png)